Modules/Sources/WordPressShared/Utility/LanguageModelHelper.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+
+#if canImport(FoundationModels)
+import FoundationModels
+#endif
+
+public enum LanguageModelHelper {
+    public static var isSupported: Bool {
+        guard #available(iOS 26, *) else { return false }
+        switch SystemLanguageModel.default.availability {
+        case .available:
+            return true
+        case .unavailable(let reason):
+            switch reason {
+            case .appleIntelligenceNotEnabled, .modelNotReady:
+                return true
+            case .deviceNotEligible:
+                return false
+            @unknown default:
+                return false
+            }
+        }
+    }
+
+    public static func makeGenerateExcerptPrompt(
+        content: String,
+        length: GeneratedContentLength,
+        style: GenerationStyle
+    ) -> String {
+        """
+        Task: Create exactly 3 excerpts for a blog post.
+
+        CRITICAL CONSTRAINTS:
+        • Each excerpt MUST be \(length.promptModifier) (\(length.wordRange) words)
+        • Style: \(style.promptDescription)
+
+        EXCERPT REQUIREMENTS:
+        * Follow the best practices for post excerpts esteblished in the WordPress ecosystem
+        • First sentence: Strong hook that creates curiosity
+        • Include the post's main value proposition
+        • Use active voice (avoid "is", "are", "was", "were" when possible)
+        • End with implicit promise of more information
+        • Do not use ellipsis (...) at the end
```

Comment:
The generated result is English when the `content` is non-English. Adding a requirement here saying that the excerpt should be in the same language as the source content?

Reply:
I wasn't going to test different locales but ran out of time.

If the post content is not supported, it will show this:

<img width="360" alt="Screenshot 2025-09-18 at 8 20 34 AM" src="https://github.com/user-attachments/assets/3e3e3079-933a-446e-819f-59f27386f70d" />

I can't get it to produce the results in anything other than English. I opened a ticket –https://linear.app/a8c/issue/CMM-762/excerpts-are-created-in-the-system-language-not-the-content-language. Help is welcome.